api/utils/rights.js

```diff
@@ -345,15 +350,30 @@ exports.validateUser = function(params, callback, callbackParam) {
 
                 params.member = member;
 
-                if (plugins.dispatch("/validation/user", {params: params})) {
-                    if (!params.res.finished) {
-                        common.returnMessage(params, 401, 'User does not have permission');
-                        reject('User does not have permission');
-                    }
-                    return false;
-                }
+                // The populator creates push messages for one app so params.qstring.apps[0] should be fine in this case
```

Comment:
Let's not do validation for app lock on `validateUser` as it has nothing to do with app context and app_id should never be passed here.

Reply:
Well, the populator uses a lot of plugins that use `validateUserForWriteAPI` which is just `validateUser` (https://github.com/Countly/countly-server/blob/next/api/utils/requestProcessor.js#L19). Here are a few examples:
- https://github.com/Countly/countly-server/blob/next/plugins/push/api/api.js#L122
- https://github.com/Countly/countly-server/blob/next/plugins/star-rating/api/api.js#L16

In fact, nothing uses `validateAppForWriteAPI` besides `/i/` and `/i/bulk` inside `requestProcessor.js` so I assume it's for sdk requests only. If so, we need an app write validator for non-sdk create/update requests and add it to wherever needed (most CE and EE plugins) or just hack some app validation into the validator most stuff is already using :smile: